pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_TNCuda_Param.cpp

```diff
@@ -0,0 +1,1579 @@
+
+        const std::vector<Precision_T> angles = {0.3};
+
+        // Results collected from `default.qubit`
+        std::vector<std::vector<cp_t>> expected_results{
+            std::vector<cp_t>(1 << num_qubits, {0.17479168, 0.02641717}),
+            std::vector<cp_t>(1 << num_qubits, {0.17479168, 0.02641717}),
+
+        };
+
+        expected_results[0][6] = {0.14837450, 0.0};
+        expected_results[0][7] = {0.14837450, 0.0};
+        expected_results[0][24] = {0.20120886, 0.0};
+        expected_results[0][25] = {0.20120886, 0.0};
+
+        expected_results[1][3] = {0.14837450, 0.0};
+        expected_results[1][7] = {0.14837450, 0.0};
+        expected_results[1][24] = {0.20120886, 0.0};
+        expected_results[1][28] = {0.20120886, 0.0};
+
+        if (inverse) {
+            for (auto &vec : expected_results) {
+                for (auto &val : vec) {
+                    val = std::conj(val);
+                }
+            }
+
+            std::swap(expected_results[0][6], expected_results[0][24]);
+            std::swap(expected_results[0][7], expected_results[0][25]);
+            std::swap(expected_results[1][3], expected_results[1][24]);
+            std::swap(expected_results[1][7], expected_results[1][28]);
+        }
+
+        SECTION("Apply adjacent wires") {
+
+            tn_state->reset();
+
+            tn_state->applyOperations(
+                {"Hadamard", "Hadamard", "Hadamard", "Hadamard", "Hadamard"},
+                {{0}, {1}, {2}, {3}, {4}}, {false, false, false, false, false});
+
+            tn_state->applyOperation("DoubleExcitationPlus", {0, 1, 2, 3},
+                                     inverse, angles);
+
+            if constexpr (std::is_same_v<TNDevice_T, MPSTNCuda<double>> ||
+                          std::is_same_v<TNDevice_T, MPSTNCuda<float>>) {
+                tn_state->append_mps_final_state();
+            }
+
+            auto results = tn_state->getDataVector();
+
+            CHECK(results == Pennylane::Util::approx(expected_results[0]));
+        }
+
+        SECTION("Apply non-adjacent wires") {
+            tn_state->reset();
+
+            tn_state->applyOperations(
+                {"Hadamard", "Hadamard", "Hadamard", "Hadamard", "Hadamard"},
+                {{0}, {1}, {2}, {3}, {4}}, {false, false, false, false, false});
+            tn_state->applyOperation("DoubleExcitationPlus", {0, 1, 3, 4},
+                                     inverse, angles);
+
+            if constexpr (std::is_same_v<TNDevice_T, MPSTNCuda<double>> ||
+                          std::is_same_v<TNDevice_T, MPSTNCuda<float>>) {
+                tn_state->append_mps_final_state();
+            }
+
+            auto results = tn_state->getDataVector();
+
+            CHECK(results == Pennylane::Util::approx(expected_results[1]));
+        }
+    }
+}
+
+TEMPLATE_LIST_TEST_CASE("TNCuda::applyMPO::SingleExcitation", "[TNCuda_Param]",
+                        TestTNBackends) {
```

Comment:
Do we want to loop exactTN as well?  Could we move the `mpo` tests to a new file?

Reply:
If we loop over `exaTN` too, we have a double check for`mpo`. I will move to a new file but will take time.

Reply:
I meant exactTN does not support MPO. We shouldn't test against it.

Reply:
Done! independent test for MPO